handlers/secrets_test.go

```diff
@@ -0,0 +1,224 @@
+package handlers
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/openfaas/faas/gateway/requests"
+	apiv1 "k8s.io/api/core/v1"
+	v1beta1 "k8s.io/api/extensions/v1beta1"
+)
+
+func Test_UpdateSecrets(t *testing.T) {
+	t.Run("No volume added if request secrets is nil", func(t *testing.T) {
+		request := requests.CreateFunctionRequest{
+			Service: "testfunc",
+			Secrets: nil,
+		}
+		existingSecrets := map[string]*apiv1.Secret{
+			"pullsecret": {Type: apiv1.SecretTypeDockercfg},
+			"testsecret": {Type: apiv1.SecretTypeOpaque, Data: map[string][]byte{"filename": []byte("contents")}},
+		}
+
+		deployment := &v1beta1.Deployment{
+			Spec: v1beta1.DeploymentSpec{
+				Template: apiv1.PodTemplateSpec{
+					Spec: apiv1.PodSpec{
+						Containers: []apiv1.Container{
+							{Name: "testfunc", Image: "alpine:latest"},
+						},
+					},
+				},
+			},
+		}
+		err := UpdateSecrets(request, deployment, existingSecrets)
+		if err != nil {
+			t.Errorf("unexpected error %s", err.Error())
+		}
+
+		validateEmptySecretVolumesAndMounts(t, deployment)
+
+	})
+
+	t.Run("No volume added if request secrets is the empty slice", func(t *testing.T) {
```

Comment:
Why are these all sub-tests?

Reply:
I was originally going to refactor out some of the common setup and possibly try to turn it into a proper table of tests

Reply:
@alexellis  I just pushed an update that I think cleans up the test names